### DISCUSSION TITLE
⚠️ SWIG 4 required for YARP master

### DISCUSSION THREAD

**Discussion by drdanz** [robotology/community#556] [Announcements]
Hello all,

I just merged robotology/yarp#2720, which means that from now on, YARP master, (and YARP 3.6 when it will be released) will require SWIG 4 to build the bindings. This was necessary, because the builds on platforms using SWIG 3.12 were failing after the adoption of C++17.

Ubuntu 18.04 and Debian 10 (which officially we still support) do not have SWIG 4 by default.

If you need YARP bindings on these platform you can install SWIG 4 from the [robotology PPA](https://launchpad.net/~robotology/+archive/ubuntu/ppa/) (Ubuntu 18.04 only), from conda, or from source.

**Comment by pattacini** [comment]
cc @robotology/everyone
